File: Practicas-121/Practica_2/linea.py
class Punto:
    def __init__(self, x, y):
        self.x, self.y = x, y

    def __str__(self):
        return f"({self.x}, {self.y})"

class Linea:
    def __init__(self, x1, y1, x2, y2):
        self.p1, self.p2 = Punto(x1, y1), Punto(x2, y2)

    def __str__(self):
        return f"Línea de {self.p1} a {self.p2}"

    def dibujar(self):
        print(f"Dibujando {self}")

if __name__ == "__main__":
    linea = Linea(0, 0, 5, 5)
    print(linea)
    linea.dibujar()

File: Practicas-121/Practica_2/linea.java
public class Linea {
    static class Punto {
        float x, y;
        Punto(float x, float y) { this.x = x; this.y = y; }
        public String toString() { return "(" + x + ", " + y + ")"; }
    }

    Punto inicio, fin;

    Linea(float x1, float y1, float x2, float y2) {
        inicio = new Punto(x1, y1);
        fin = new Punto(x2, y2);
    }

    public String toString() { return "Línea de " + inicio + " a " + fin; }

    void dibujar() { System.out.println("Dibujando " + this); }

    public static void main(String[] args) {
        Linea linea = new Linea(1, 2, 3, 4);
        System.out.println(linea);
        linea.dibujar();
    }
}
